RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc

```diff
@@ -48,61 +48,157 @@ namespace {
   inline int nPixelHits(reco::Track const & tk) {
     return tk.hitPattern().numberOfValidPixelHits();
   }
-      
+  
   inline float dz(reco::Track const & trk, Point const & bestVertex) {
-      return std::abs(trk.dz(bestVertex));
+    return std::abs(trk.dz(bestVertex));
   }
   inline float dr(reco::Track const & trk, Point const & bestVertex) {
-      return std::abs(trk.dxy(bestVertex));
+    return std::abs(trk.dxy(bestVertex));
+  }
+
+  inline void dzCut_par1(reco::Track const & trk, int & nLayers, const float * par, float dzCut[]) {
+    float dzE =  trk.dzError();
+    for (int i=2; i>=0; --i) {
+      dzCut[i] = powN(par[i]*nLayers,4)*dzE;
+    }
+  }
+  inline void drCut_par1(reco::Track const & trk, int & nLayers, const float* par, float drCut[]) {
+    float drE =  trk.d0Error();
+    for (int i=2; i>=0; --i) {
+      drCut[i] = powN(par[i]*nLayers,4)*drE;
+    }
+  }
+  
+  inline void dzCut_par2(reco::Track const & trk, int & nLayers, const float * par, const float * d0err, const float * d0err_par, float dzCut[]) {
+    float pt = float(trk.pt());
+    float p  = float(trk.p());
+
+    for (int i=2; i>=0; --i) {
+      // parametrized d0 resolution for the track pt
+      float nomd0E = sqrt(d0err[i]*d0err[i]+(d0err_par[i]/pt)*(d0err_par[i]/pt));
+      // parametrized z0 resolution for the track pt and eta
+      //    float nomdzE = nomd0E*(std::cosh(eta));
+      float nomdzE = nomd0E*(abs(p)/pt); // cosh(eta):=abs(p)/pt
```

Comment:
std::abs, if any "abs" at all.
p is supposed to be non-negative by construction.